examples/src/main/scala/org/apache/spark/examples/ml/BisectingKMeansExample.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.examples.ml
+
+// scalastyle:off println
+
+// $example on$
+import org.apache.spark.ml.clustering.BisectingKMeans
+import org.apache.spark.mllib.linalg.{Vectors, VectorUDT}
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+import org.apache.spark.sql.types.{StructField, StructType}
+// $example off$
+
+/**
+ * An example demonstrating a bisecting k-means clustering.
+ * Run with
+ * {{{
+ * bin/run-example ml.BisectingKMeansExample
+ * }}}
+ */
+object BisectingKMeansExample {
+
+  def main(args: Array[String]): Unit = {
+    // Creates a SparkSession
+    val spark = SparkSession.builder.appName("BisectingKMeansExample").getOrCreate()
+
+    // $example on$
+    // Crates a DataFrame
+    val rowRDD = spark.read.text("data/mllib/kmeans_data.txt").rdd
```

Comment:
Yeah, I propose we just use an existing LIBSVM example data file, or we can
create a new one from kmeans_example_data.

On Fri, 6 May 2016 at 10:08 Yanbo Liang <notifications@github.com> wrote:

> In
> examples/src/main/scala/org/apache/spark/examples/ml/BisectingKMeansExample.scala
> <https://github.com/apache/spark/pull/11844#discussion_r62299617>:
>
> > +/**
> > + * An example demonstrating a bisecting k-means clustering.
> > + * Run with
> > + * {{{
> > + * bin/run-example ml.BisectingKMeansExample
> > + * }}}
> > + */
> > +object BisectingKMeansExample {
> > +
> > +  def main(args: Array[String]): Unit = {
> > +    // Creates a SparkSession
> > +    val spark = SparkSession.builder.appName("BisectingKMeansExample").getOrCreate()
> > +
> > +    // $example on$
> > +    // Crates a DataFrame
> > +    val rowRDD = spark.read.text("data/mllib/kmeans_data.txt").rdd
>
> @sethah <https://github.com/sethah> 's proposal is feasible. I'd like to
> use dataset with libsvm format, then we can load it directly use
> spark.read.format("libsvm"). We can get features with Vector type and
> feed them into model training. Although the dataset has label column but we
> don't use actually. This will make the example more succinct.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/11844/files/e2aaabd318a76b6edc59a99cfbc0f6239c833c0c#r62299617>
>